pkg/controller/registry/grpc/source.go

```diff
@@ -100,10 +105,71 @@ func (s *SourceStore) Get(key registry.CatalogKey) *SourceConn {
 	return &source
 }
 
+func grpcProxyURL(addr string) (*url.URL, error) {
+	// Handle ip addresses
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+
+	// Hardcode fields required for proxy resolution
+	url.Host = addr
+	url.Scheme = "http"
+
+	// Override HTTPS_PROXY and HTTP_PROXY with GRPC_PROXY
+	proxyConfig := &httpproxy.Config{
+		HTTPProxy:  getGRPCProxyEnv(),
+		HTTPSProxy: getGRPCProxyEnv(),
+		NoProxy:    getEnvAny("NO_PROXY", "no_proxy"),
+		CGI:        os.Getenv("REQUEST_METHOD") != "",
```

Comment:
We should probably memoize these so we don't fetch them for every dial.

Reply:
This seems reasonable but would make the values static, which likely isn't a problem given how deployments and pods work.